src/NLSModels.jl

```diff
@@ -183,14 +184,24 @@ function jac_op_residual!(nls :: AbstractNLSModel, x :: AbstractVector,
 end
 
 """
-    Hi = hess_residual(nls, x, i)
+    Hi = hess_residual(nls, x, v)
 
-Computes the Hessian of the i-th residual at x.
+Computes the linear combinations of the Hessians of the residuals at x with coefficients
+`v`.
 """
-function hess_residual(nls :: AbstractNLSModel, x :: AbstractVector, i :: Int)
+function hess_residual(nls :: AbstractNLSModel, x :: AbstractVector, v :: AbstractVector)
   throw(NotImplementedError("hess_residual"))
 end
 
+"""
+    Hi = jth_hess_residual(nls, x, i)
```

Comment:
Maybe use `j` instead of `i`?

Reply:
I've noticed that I've used `i` in a lot of places for Hessian related things, though `jth` existed for other things. Do you think it's better to update all to `j`?

Reply:
It's a detail but it may be clearer in the docstring (in the functions, it doesn't really matter).

Reply:
Updated.